File: python/Dosen.py
# Membuat kelas baru bernama Dosen, class ini merupakan child class dari SivitasAkademik Class
from SivitasAkademik import SivitasAkademik

class Dosen(SivitasAkademik):
    
    # Inisialisasi variabel untuk setiap objek dari kelas
    nip = ""            # Membuat variabel untuk nomor induk pegawai   
    fakultas = ""       # Membuat variabel untuk fakultas tempat dosen bekerja
    pend_terakhir = ""  # Membuat variabel untuk pendidikan terakhir dosen
    keahlian = ""       # Membuat variabel untuk keahlian dosen
    
    # Konstruktor untuk inisialisasi variabel pada objek dari kelas Dosen
    def __init__(self, NIK = "", name = "", gender = "", asal_univ = "", email_edu = "", nip = "", fakultas = "", pend_terakhir = "", keahlian = ""):
        super().__init__(NIK, name, gender, asal_univ, email_edu)
        self.nip = nip                      # Menginisialisasi variabel nip
        self.fakultas = fakultas            # Menginisialisasi variabel fakultas
        self.pend_terakhir = pend_terakhir  # Menginisialisasi variabel pend_terakhir
        self.keahlian = keahlian            # Menginisialisasi variabel keahlian
        
    # Getter dan Setter untuk NIP
    def get_Nip(self):
        return self.nip
    
    def set_Nip(self, nip):
        self.nip = nip
        
    # Getter dan Setter untuk Fakultas
    def get_Fakultas(self):
        return self.fakultas
    
    def set_Fakultas(self, fakultas):
        self.fakultas = fakultas
        
    # Getter dan Setter untuk Pend_terakhir
    def get_Pend_terakhir(self):
        return self.pend_terakhir
    
    def set_Pend_terakhir(self, pend_terakhir):
        self.pend_terakhir = pend_terakhir
        
    # Getter dan Setter untuk Keahlian
    def get_Keahlian(self):
        return self.keahlian
    
    def set_Keahlian(self, keahlian):
        self.keahlian = keahlian
        

File: cpp/Dosen.cpp
class Dosen :  public SivitasAkademik{
    
    /*
    Saya Talitha Syahla NIM 2101330 mengerjakan Soal Latihan 4 
    dalam mata kuliah Desain Pemrograman Berorientasi Objek 
    untuk keberkahanNya maka saya tidak melakukan kecurangan 
    seperti yang telah dispesifikasikan. Aamiin.
    */

    private:
        string nip;
        string fakultas;
        string pend_terakhir;
        string keahlian;

    public:
    // Constructor
        Dosen(){
            this->nip = "";
            this->fakultas = "";
            this->pend_terakhir = "";
            this->keahlian = "";
        }

        Dosen(string NIK, string name, string gender, string asal_univ, string email_edu, string nip, string fakultas, string pend_terakhir, string keahlian) : SivitasAkademik(NIK, name, gender, asal_univ, email_edu){
            this->nip = nip;
            this->fakultas = fakultas;
            this->pend_terakhir = pend_terakhir;
            this->keahlian = keahlian;
        }

    // Setter dan Getter

        // Untuk nip
        string get_Nip(){
            return this->nip;
        }

        void set_Nip(string nip){
            this->nip = nip;
        }

        // Untuk Fakultas
        string get_Fakultas(){
            return this->fakultas;
        }

        void set_Fakultas(string fakultas){
            this->fakultas = fakultas;
        }

        // Untuk Pend_terakhir
        string get_Pend_terakhir(){
            return this->pend_terakhir;
        }

        void set_Pend_terakhir(string pend_terakhir){
            this->pend_terakhir = pend_terakhir;
        }

        // Untuk Keahlian
        string get_Keahlian(){
            return this->keahlian;
        }

        void set_Keahlian(string keahlian){
            this->keahlian = keahlian;
        }

        // Destruktor
        ~Dosen(){
        }
    
};